### ISSUE TITLE
Restoring not working properly

### ISSUE THREAD

**Issue by pheroh** [TeamWin/Team-Win-Recovery-Project#24]
I have written my case here along with someone else's:
http://forum.xda-developers.com/showthread.php?p=29384836

I was using v2.2.0 and was testing Sense based ROMS.

Device is HTC One S, S4 processor.
My HBOOT is 1.13 and I did flash the kernel via fastboot as TWRP can't flash it.

**Comment by pheroh** [comment]
Just to have the whole problem here, here is the text from the link:
I faced a similar problem and I freaked out. I installed MIUI (Sense-base) to test it and then when I wanted to revert back and after trying to restore my Trickdroid nandroid, every time I boot the phone after restoring (btw, it took the normal time to restore unlike OP), it kept asking if I want to use the miui launcher and then an error appeared saying your phone has stopped and nothing works. It seems that some leftover was still present from the MIUI rom.

I did wipe everything (system, data, cache, dalvik) before and after restoring but the same thing happened over and over. It also refused to install any new ROM. I thought I had a brick. Then after a couple of reboots to recovery, I could install a fresh trickdroid and then on top of it I could restore my nandroid.

Now I am weary of installing anything new. I have a CWM nandroid as well just in case TWRP does that again

**Comment by bigbiff** [comment]
Please send us /tmp/recovery.log from recovery so we can see what's going on. You can also get this log by going to Advanced->Copy Log to SDCard. In android you can get the recovery.log under /sdcard and paste it into paste.tinyw.in. Please paste the URL in here.

**Comment by pheroh** [comment]
https://paste.tinyw.in/index.php/view/42651480
Not sure this is helpful since it seems to only have the fresh install of trickdroid after it failed to restore it properly.. It also doesn't show the immediate restore I did after the fresh install.

Just to add. I had the same experience again yesterday with CM9 rom this time and restoring didn't work and I had to reinstall a fresh one (the one in the log).

If it might be of any help, I also flashed the boot.img (via fastboot) before restoring the nandroid and wiping everything.. Not sure if the kernel has anything to do with wiping but I thought I'd share that.

**Comment by Dees-Troy** [comment]
We'd like a log from a failed restore.  In other words, restore a backup and copy the log.  Then double-check the failed restore by booting up the device after finishing the restore and copying the log.  The log you provided just has the zip install and no restore.

**Comment by pheroh** [comment]
Here is a log of a failed restore. I confirmed that the restore doesn't work properly.
https://paste.tinyw.in/index.php/view/51780450

Edit: here is another log of another restore that also failed once the ROM boots. This restore is the one I did on top of the failed one above.
https://paste.tinyw.in/index.php/view/64133173

**Comment by pheroh** [comment]
Update!
I downgraded to 2.1.8 and tried restoring the same nandroid (paranoid2) and it worked flawlessly!
Log: http://tinyw.in/SxKF

Edit: Actually, again the problem persists :(
After the successful paranoid2 restore, I tied to restore my Trickdroid daily driver and the ROM failed.. It boots (just like with 2.2.1) but then an error message says something went wrong with system process and I have an unresponsive device. Unfortunately, I forgot to copy the log of the restore.

Edit 2: When restoring, I don't restore boot but I flash it via fastboot instead. paranoid2 has the boot backup but I didn't restore it in the process but rather via fastboot. However, in Trickdroid nandroid I didn't backup the boot to begin with.. Could this be the problem? Still, I DID flash the boot in fastboot.

**Comment by pheroh** [comment]
Update 2.
I did a full backup this time including boot and cache and I restored everything except boot (mimicking paranoid2 restore except paranoid2 had no cache backup) and it worked and I had a proper restore.

I also did fix permission after flashing the kernel.. Maybe this helped? I dunno..

Restore log: http://tinyw.in/gdkG

Nonetheless, it does seem that 2.2.x has something broken as paranoid2 didn't work with the same procedure there.
